src/wp-includes/class-wp-customize-panel.php

```diff
@@ -235,7 +235,7 @@ public function json() {
 	 *
 	 * @return bool False if theme doesn't support the panel or the user doesn't have the capability.
 	 */
-	final public function check_capabilities() {
+	public function check_capabilities() {
```

Comment:
Had to mark this as non `final` which I think makes sense to do anyway.

Reply:
> Had to mark this as non `final` which I think makes sense to do anyway.

@westonruter Do you recall if there was an essential reason `WP_Customize_Panel::check_capabilities()` was marked `final`?

Reply:
I don't see any discussion about `final` in the original tickets that added this code, which makes me think it was just done in an attempt to avoid backwards compatibility issues while this new API was being iterated on. It's a stable API now (committed 7 years ago) so I don't think we need to worry about that anymore.

https://core.trac.wordpress.org/ticket/29197

An alternative is that we could add a `$theme_supports_callback` property which would work similarly to the existing `$active_callback` that is used by `active()` which is marked final. I'm not such a fan of this though because then it isn't clear what should happen when both `$theme_supports` and `$theme_supports_callback` is set. Allowing `check_capabilities` to be overridden feels simpler and more in line with the Customizer's OOP design to me.

Reply:
> This might need a dev note to let extenders know that this is now available to them. What do you think?

I think it's okay because there was already `$theme_supports` which allows just this. Overriding `check_capabilities()` is for more complex use-cases.

Maybe at most a little note in the "miscellaneous API changes" dev note is okay.

Reply:
Yeah, I went looking for a hint in Slack and IRC but couldn't find anything either. 

An alternative may be to allow `this->capability` to be `string|string[]` and looping through.

Reply:
A little more on the history of the `final` keyword here:

The `final` keyword was added to `WP_Customize_Section` and `WP_Customize_Setting` in changeset [20248](https://core.trac.wordpress.org/changeset/20248).

The only direct reference to `check_capabilities()` is:
> prepare_controls() now removes any settings and sections that return false for check_capabilities().

Then, in changeset [29487](https://core.trac.wordpress.org/changeset/29487), `WP_Customize_Panel` was separated to its own file. Having previously extended `WP_Customize_Section`, the change included copying the `check_capabilities()` method into `WP_Customize_Panel`.

This was likely just a copy-paste for consistency.

Reply:
> An alternative may be to allow `this->capability` to be `string|string[]` and looping through.

Unfortunately we're already using `string[]` for additional arguments that should be passed into `current_theme_supports`.